wp-content/civi-extensions/goonjcustom/Civi/InductionService.php

```diff
@@ -402,6 +402,11 @@ public static function hasIndividualChangedToVolunteer($op, $objectName, $id, &$
     if ($op !== 'edit' || $objectName !== 'Individual') {
       return FALSE;
     }
+    // Check if 'entryURL' is set and contains '/contribute'.
+    if (isset($params['entryURL']) && strpos($params['entryURL'], '/contribute') !== FALSE) {
+      // Return if '/contribute' is found in the entryURL.
+      return FALSE;
+    }
```

Comment:
why do we need this?

Reply:
If any contact is already on DB and it's a volunteer then I was unable to make a payment some error I got there this function stopped me from moving further. By adding this check I'm return from this function because I don't want to call this function while doing the payment.